nomad/rpc.go

```diff
@@ -266,7 +266,56 @@ func (s *Server) handleStreamingConn(conn net.Conn) {
 // using the Yamux multiplexer. Version 2 handling allows a single connection to
 // switch streams between regulars RPCs and Streaming RPCs.
 func (s *Server) handleMultiplexV2(conn net.Conn, ctx *RPCContext) {
-	// TODO
+	defer func() {
+		// Remove any potential mapping between a NodeID to this connection and
+		// close the underlying connection.
+		s.removeNodeConn(ctx)
+		conn.Close()
+	}()
+
+	conf := yamux.DefaultConfig()
+	conf.LogOutput = s.config.LogOutput
+	server, _ := yamux.Server(conn, conf)
```

Comment:
Looks like you can ignore the error because it only errors on an invalid config and there's no risk of that?

Still a bit worried about future changes affecting either of these constraints (either more config changes going unvalidated _or_ new errors returned by `Server()`)

Maybe check, log an error and exit now just to make the code more resilient to surrounding code changes? Not a blocker.

Reply:
Added it to both call sites.